### PR TITLE
Refactor/comscore cleanup

### PR DESCRIPTION
**Pull request by therama**

### CODE REVIEW

File: Code/Comscore/Source/integration/ComscoreAdapter.swift [hunk 1]
```diff
@@ -474,22 +475,18 @@ class THEOComScoreAdapter: NSObject {
...
-                    let dvrWindowEnd = lastRange.end
-                    let dvrWindowStart = firstRange.start
-                    let dvrWindowLengthInSeconds = dvrWindowEnd - dvrWindowStart
-                    if (dvrWindowLengthInSeconds > 0) {
-                        if welf.configuration.debug { print("[THEOplayerConnectorComscore] set DVR window length of ",dvrWindowLengthInSeconds) }
-                        welf.streamingAnalytics.setDVRWindowLength(Int(dvrWindowLengthInSeconds*1000))
-                    }
-                }
-            })
+        if self.comscoreMetadata.length != 0 || self.inAd {
```
Comment (therama) [inline Code/Comscore/Source/integration/ComscoreAdapter.swift:478]: This looks strange to me. Can anyon explain why we return early when comscoreMetadata.length is 0? (same for `onSeeked`)

Comment (wjoosen) [reply Code/Comscore/Source/integration/ComscoreAdapter.swift:478]: In that event handler the DVR window length should be set if the current source is a live stream. 

-  If  `comscoreMetadata.length == 0` , it's a live stream (see [ComscoreMetadata.length](https://github.com/THEOplayer/iOS-Connector/blob/7b2b9b6d0ce1f9dd20d2daffe5d528215a053eb4/Code/Comscore/Source/integration/ComscoreMetadata.swift#L16)). I assume this is something from an old customer implementation. Doing the check with `comscoreMetadata.mediaType == .live` would make more sense nowadays and is better for readability.
- If `self.inAd == true`, the seekable range for the ad would be returned

**Review by wjoosen — APPROVED** [review]

File: Code/Comscore/Source/integration/ComscoreAdapter.swift (hunk 1 above)
Comment (therama) [reply Code/Comscore/Source/integration/ComscoreAdapter.swift]: Thanks, I updated to use `mediaType` instead

File: Code/Comscore/Source/integration/ComscoreAdapter.swift [hunk 2]
```diff
@@ -444,10 +446,9 @@ class THEOComScoreAdapter: NSObject {
         if configuration.debug { print("[THEOplayerConnectorComscore] AD_BEGIN event") }
         if let ad = event.ad {
             currentAdDuration = 0
-            if let duration = player.duration {
-                if duration != .infinity && duration != .nan {
-                    currentAdDuration = Int(duration * 1000)
-                }
+            if let linearAd = ad as? LinearAd,
+               let duration: Int = linearAd.duration {
+                self.currentAdDuration = duration
```
Comment (wvanhaevre) [inline Code/Comscore/Source/integration/ComscoreAdapter.swift:451]: Just to make sure: currentAdDuration was previously converted to msec? Is that also msec on the linearad? Or maybe that was a mistake?

**Review by wvanhaevre — APPROVED** [review]

File: Code/Comscore/Source/integration/ComscoreAdapter.swift (hunk 2 above)
Comment (therama) [reply Code/Comscore/Source/integration/ComscoreAdapter.swift:451]: Good catch! I will convert it to an msec.

**Review by Danesz — APPROVED** [review]

**Pull request merged**